airbyte-integrations/connectors/source-elasticsearch/src/main/java/io/airbyte/integrations/source/elasticsearch/ElasticsearchConnection.java

```diff
@@ -146,11 +146,18 @@ private RuntimeException unwrappedApiException(String message, ApiException e) {
    * @throws IOException throws IOException if Elasticsearch request fails
    */
   public Map<String, MappingMetadata> getMappings(final List<String> indices) throws IOException {
-    GetMappingsRequest request = new GetMappingsRequest();
-    String[] copiedIndices = indices.toArray(String[]::new);
-    request.indices(copiedIndices);
-    GetMappingsResponse getMappingResponse = client.indices().getMapping(request, RequestOptions.DEFAULT);
-    return getMappingResponse.mappings();
+    int chunk = 15;
```

Comment:
@jeromedoucet what about moving this value to a config parameter? from where did you choose it?

Reply:
@marcosmarxm 

> what about moving this value to a config parameter ?

It should not.

Config parameters are supposed to be user specific. This isn't the case here. This is just a programming error fix, a bug (bad `getMapping()` usage in some circumstances), not a performance tuning parameter or something that change the functional behavior. Users should not be able to change that.

That's true that users may want to have more control over how requests are done against there ES cluster, but this is a different story in which the way data are fetch are involved too (for instance).

> from where did you choose it ?

That's based on some tests from our ES log cluster, trying to figure out a reasonable (It may be a little bit more conservative :shrug: ) index number limit that will prevent the 4096 bytes limits url size.

A more precise approach is trying to compute the chunk size dynamically, based on indexes name size + other URL element size, but : 

- It Depends on ES client implementation details, and that's wrong
- It's more complex, then more difficult to understand and maintain for few advantages

There is something we may add to this implementation. If, despite this chuck mechanism, the 4096 bytes limits url size is reached (if an index has a verrrrry long name for instance), we may catch the exception and retry with fewer indices. But I am unsure it's really useful